### PR TITLE
Added feature to restrict the limit to a maximum

### PR DESCRIPTION
**Pull request by deviouspk**

### CODE REVIEW

File: Abstracts/Repositories/Repository.php [hunk 1]
```diff
@@ -18,6 +18,8 @@
 {
 
     use PrettusCacheableRepository;
+    
+    protected $maxPaginationLimit;
```
Comment (johannesschobel) [inline Abstracts/Repositories/Repository.php]: shall we set this to `null` by default?

Comment (deviouspk) [reply Abstracts/Repositories/Repository.php]: Isn't it automatically null if you don't define it?

$maxPaginationLimit = null equals $maxPaginationLimit;

Comment (johannesschobel) [reply Abstracts/Repositories/Repository.php]: yeah, but i think, that `isset($this->maxPaginationLimit)` and `empty($this->maxPaginationLimit)` behave different if the value is not set - but not sure on this one..

Or shall we just set the `maxPaginationLimit` by default to `0` to "disable" it.. 
And the `if-statement` down below should check, if `$this->maxPaginationLimit` is an integer and `>0`

what do you think?!

Comment (deviouspk) [reply Abstracts/Repositories/Repository.php]: added the changes you proposed with 0 as default value :) Will you add documentation please?

**Comment by johannesschobel** [comment]
Hey there.. thank you so much for the PR and changes.. I will look into it later today and merge! Will give feedback shortly!

**Comment by johannesschobel** [comment]
thanks a lot!

**Pull request merged**